data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/S3Source.java

```diff
@@ -13,44 +13,57 @@
 import com.amazon.dataprepper.model.record.Record;
 import com.amazon.dataprepper.model.source.Source;
 import org.apache.commons.lang3.NotImplementedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
 import software.amazon.awssdk.regions.Region;
 import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.sqs.SqsClient;
 import software.amazon.awssdk.services.sts.StsClient;
 
 import java.util.Objects;
 
 @DataPrepperPlugin(name = "s3", pluginType = Source.class, pluginConfigurationType = S3SourceConfig.class)
 public class S3Source implements Source<Record<Event>> {
+    private static final Logger LOG = LoggerFactory.getLogger(S3Source.class);
 
     private final PluginMetrics pluginMetrics;
     private final S3SourceConfig s3SourceConfig;
+    private AwsCredentialsProvider awsCredentialsProvider;
+    private S3Client s3Client;
+    private SqsClient sqsClient;
 
     @DataPrepperPluginConstructor
     public S3Source(PluginMetrics pluginMetrics, final S3SourceConfig s3SourceConfig) {
         this.pluginMetrics = pluginMetrics;
         this.s3SourceConfig = s3SourceConfig;
 
         awsCredentialsProvider = createCredentialsProvider();
+        LOG.info("Creating S3 client.");
         s3Client = createS3Client(awsCredentialsProvider);
+        sqsClient = SqsClient.create();
 
         throw new NotImplementedException();
     }
 
     @Override
     public void start(Buffer<Record<Event>> buffer) {
+        if (buffer == null) {
+            throw new IllegalStateException("Buffer provided is null");
+        }
 
+        for(int i = 0; i < s3SourceConfig.getSqsOptions().getThreadCount(); i++) {
+            Thread sqsWorkerThread = new Thread(new SqsWorker(sqsClient, s3Client, s3SourceConfig));
```

Comment:
I think we will want to have a `List<Thread>` as a field. You can keep all the threads there. You will need to stop them when the `stop()` method for S3Source is called.